store/views.py

```diff
@@ -4,79 +4,122 @@
+                else: 
+                    bookdata = Book.objects.filter(genre=get_data['genre'],author=get_data['author'],title=get_data['title'])
+            
     context = {
-        'books': None, # set this to the list of required books upon filtering using the GET parameters
-                       # (i.e. the book search feature will also be implemented in this view)
+        'books': bookdata,
     }
-    get_data = request.GET
-    # START YOUR CODE HERE
-    
-    
     return render(request, template_name, context=context)
 
 @login_required
 def viewLoanedBooks(request):
     template_name = 'store/loaned_books.html'
+    if request.user.is_authenticated:
+        books = BookCopy.objects.filter(borrower=request.user)
     context = {
-        'books': None,
+        'books': books,
     }
-    '''
-    The above key 'books' in the context dictionary should contain a list of instances of the 
-    BookCopy model. Only those book copies should be included which have been loaned by the user.
-    '''
-    # START YOUR CODE HERE
-    
-
-
     return render(request, template_name, context=context)
 
 @csrf_exempt
 @login_required
 def loanBookView(request):
-    response_data = {
-        'message': None,
-    }
-    '''
-    Check if an instance of the asked book is available.
-    If yes, then set the message to 'success', otherwise 'failure'
-    '''
-    # START YOUR CODE HERE
-    book_id = None # get the book id from post data
-
-
+    book_id = request.POST.get('bid')
+    availBooks=BookCopy.objects.filter(book=Book.objects.get(id=book_id),status=True)
+    print(len(availBooks))
+    if len(availBooks)!=0:
+        booktoget = availBooks.first()
+        booktoget.status=False
+        booktoget.borrow_date=date.today()
+        booktoget.borrower=request.user
+        booktoget.save(update_fields=['status','borrow_date','borrower'])
+        response_data = {
+            'message': 'success',
+        }
+    else:
+        response_data = {
+            'message': 'failure',
+        }
     return JsonResponse(response_data)
 
-'''
-FILL IN THE BELOW VIEW BY YOURSELF.
-This view will return the issued book.
-You need to accept the book id as argument from a post request.
-You additionally need to complete the returnBook function in the loaned_books.html file
-to make this feature complete
-''' 
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    bcid = request.POST.get('bcid')
+    booktoret = BookCopy.objects.get(id=bcid)
+    booktoret.status=True
+    booktoret.borrower = None
+    booktoret.borrow_date=None
+    booktoret.save(update_fields=['status','borrower','borrow_date'])
+    response_data = {
+            'message': 'success',
+        }
+    return JsonResponse(response_data)
 
 
+def rateBook(request):
+    bid = request.POST.get('bid')
+    rating = request.POST.get('rating')
+    ratingbook=Book.objects.get(id=bid)
```

Comment:
Though the validation of rating is present in the models, it will raise an error during save as you haven't added any validation here.

Also, just after rating the book, the value isn't updated in the DOM. You could've atleast redirected to some other page, or have reloaded the page.